src/ExceptionTransformer.ts

```diff
@@ -46,9 +48,80 @@ class ExceptionTransformer {
         exception.fallback_message
       );
     }
-
     return exceptionMap;
   }
+
+  public generateSpesificFieldError(errorInfo: Exception | null | undefined) {
+    const detail =
+      errorInfo && typeof errorInfo.detail === "object" && errorInfo.detail;
+
+    if (detail) {
+      return function getError(fieldName: string) {
+        return detail[fieldName];
+      };
+    } else {
+      return () => {};
+    }
+  }
+
+  public generateErrorMessage(
+    errorInfo: Exception,
+    options: Options = {}
+  ): string {
+    let finalMessage = "";
+    const { knownErrorKeys = [], skipTypes = [] } = options;
+
+    try {
+      if (!(skipTypes && skipTypes.includes(errorInfo.type))) {
+        const errorDetail = errorInfo.detail;
+        let shouldDisplayFallbackMessage = false;
+        let message = "";
+
+        if (errorDetail && typeof errorDetail === "object") {
+          const errorDetailKeys = Object.keys(errorDetail);
+
+          if (
+            errorDetailKeys.includes("non_field_errors") &&
+            errorDetail.non_field_errors
+          ) {
+            if (typeof errorDetail.non_field_errors[0] === "string") {
+              message = errorDetail.non_field_errors[0];
+            } else {
```

Comment:
I will open an issue about this to extract a logical error message from `errorDetail.non_field_errors[0]` if it exists. So in your first if block, the condition should just check if `errorDetail.non_field_errors[0]` is null or undefined and then we will recursively look for a value. We can publish without making this enhancement but it will be a good addition.